src/derivative_operators/ghost_derivative_operator.jl

```diff
@@ -1,74 +1,39 @@
-struct GhostDerivativeOperator{T<:Real, E<:AbstractDerivativeOperator{T}, F<:AbstractBC{T}} <: AbstractDiffEqLinearOperator{T}
+struct GhostDerivativeOperator{T, E<:AbstractDiffEqLinearOperator{T}, F<:AbstractBC{T}} <: AbstractDiffEqLinearOperator{T}
     L :: E
     Q :: F
 end
 
-function *(L::AbstractDerivativeOperator{T}, Q::AbstractBC{T}) where{T}
+function *(L::AbstractDiffEqLinearOperator{T}, Q::AbstractBC{T}) where{T}
     return GhostDerivativeOperator{T, typeof(L), typeof(Q)}(L,Q)
 end
 
 function *(L::AbstractDiffEqCompositeOperator{T}, Q::AbstractBC{T}) where{T}
     return sum(map(op -> op * Q, L.ops))
 end
 
-function LinearAlgebra.mul!(x::AbstractVector, A::GhostDerivativeOperator, u::AbstractVector)
-    @assert length(u) == A.L.len == length(x)
+function *(A::GhostDerivativeOperator{T1}, u::AbstractArray{T2}) where {T1,T2}
+    #TODO Implement a function domaincheck(L::AbstractDiffEqLinearOperator, u) to see if components of L along each dimension match the size of u
+    x = zeros(promote_type(T1,T2), unpadded_size(u))
     LinearAlgebra.mul!(x, A.L, A.Q*u)
-end
-
-function *(A::GhostDerivativeOperator{T1}, u::AbstractVector{T2}) where {T1,T2}
-    @assert length(u) == A.L.len
-    x = zeros(promote_type(T1,T2), A.L.len)
-    LinearAlgebra.mul!(x, A, u)
     return x
 end
 
-function LinearAlgebra.mul!(M_temp::AbstractMatrix, A::GhostDerivativeOperator, M::AbstractMatrix)
-    @assert size(M,1) == size(M_temp,1) == A.L.len
-    for i in 1:size(M,2)
-        mul!(view(M_temp,:,i), A, view(M,:,i))
-    end
-end
-
-function *(A::GhostDerivativeOperator{T1}, M::AbstractMatrix{T2}) where {T1,T2}
-    @assert size(M,1) == A.L.len
-    M_temp = zeros(promote_type(T1,T2), A.L.len, size(M,2))
-    LinearAlgebra.mul!(M_temp, A, M)
-    return M_temp
-end
-
-
-function LinearAlgebra.ldiv!(x::AbstractVector, A::GhostDerivativeOperator, u::AbstractVector)
-    @assert length(x) == A.L.len
-    (AL,Ab) = Array(A)
-    LinearAlgebra.ldiv!(x, lu!(AL), u-Ab)
-end
```

Comment:
Where did these dispatches go?

Reply:
The first two seem to be captured fine by defaults, or at least `3D_laplacian.jl` works out fine (granted it's 3D, so maybe I'm missing why the Matrix part of this dispatch is significant).

```
julia> (typeof((A*Q).ops[1]), typeof(u0))
(GhostDerivativeOperator{Float64,DerivativeOperator{Float64,1,false,Float64,StaticArrays.SArray{Tuple{5},Float64,1,5},StaticArrays.SArray{Tuple{1},StaticArrays.SArray{Tuple{6},Float64,1,6},1,1},Nothing,Nothing},ComposedMultiDimBC{Float64,RobinBC{Float64,V} where V<:AbstractArray{Float64,1},3,2}}, Array{Float64,3})

julia> which(LinearAlgebra.mul!, (typeof(u0), typeof((A*Q).ops[1]), typeof(u0)))
mul!(Y::AbstractArray, L::DiffEqBase.AbstractDiffEqLinearOperator, B::AbstractArray) in DiffEqBase at /home/graham/.julia/packages/DiffEqBase/zXRWE/src/operators/common_defaults.jl:17

julia> which(Base.:(*), (typeof((A*Q).ops[1]), typeof(u0)))
*(A::GhostDerivativeOperator{T1,E,F} where F<:DiffEqOperators.AbstractBC{T1} where E<:DiffEqBase.AbstractDiffEqLinearOperator{T1}, u::AbstractArray{T2,N} where N) where {T1, T2} in DiffEqOperators at /home/graham/.julia/dev/DiffEqOperators/src/derivative_operators/ghost_derivative_operator.jl:14
```
where the variables are as in `3D_laplacian.jl`

Caveat:  The output plot from `3D_laplacian.jl` looks pretty, but I don't actually know what it should look like. It's basically concentric 3D waves emanating from the center of space and then bouncing back from the edges of the cubic space and interfering with each other a bit.

Reply:
I can confirm that that is what it's supposed to look like, it's the wave eqn